datafusion/proto/src/from_proto.rs

```diff
@@ -1137,10 +1138,20 @@ pub fn parse_expr(
                     parse_expr(&args[0], registry)?,
                     parse_expr(&args[1], registry)?,
                 )),
-                ScalarFunction::Substr => Ok(substr(
-                    parse_expr(&args[0], registry)?,
-                    parse_expr(&args[1], registry)?,
-                )),
+                ScalarFunction::Substr => {
+                    if args.len() > 2 {
```

Comment:
I recommend adding an assert `assert_eq!(args.len(), 3)` in this case so that we don't get another silent failure if we add additional argument support to substr

Reply:
Thanks for the feedback, I've added the assert in the new commit :)